console/console-init/ui/src/modules/iot-device/components/CreateMetadata/MetaDataRow.tsx

```diff
@@ -4,77 +4,36 @@
  */
 
 import React, { useState } from "react";
-import {
-  Grid,
-  GridItem,
-  InputGroup,
-  Button,
-  DropdownPosition,
-  TextInput
-} from "@patternfly/react-core";
-import { DropdownWithToggle } from "components";
-import { PlusIcon, MinusCircleIcon } from "@patternfly/react-icons";
-import {
-  deviceRegistrationTypeOptions,
-  IMetadataProps
-} from "modules/iot-device";
-import { DataType } from "constant";
+import { Grid, GridItem } from "@patternfly/react-core";
 import { ValidationStatusType } from "modules/iot-device/utils";
+import { isObjectOrArray } from "utils";
+import { MetaDataProperty } from "./MetaDataProperty";
+import { MetaDataType } from "./MetaDataType";
+import { MetaDataValue } from "./MetaDataValue";
 
-export interface IMetaDataRow {
+export interface IMetaDataRowProps {
   metadataList: any;
   setMetadataList: (metadataList: any) => void;
   rowIndex: number;
 }
 
-export const MetaDataRow: React.FC<IMetaDataRow> = ({
+export const MetaDataRow: React.FC<IMetaDataRowProps> = ({
   metadataList,
   setMetadataList,
   rowIndex
 }) => {
-  const currentRow = metadataList[rowIndex];
   const [validationStatus, setValidationStatus] = useState<
     ValidationStatusType.DEFAULT | ValidationStatusType.ERROR
   >(ValidationStatusType.DEFAULT);
 
   const updateMetadataList = (property: string, value: string) => {
     let updatedMetadataList = [...metadataList];
-    switch (property) {
-      case "key":
-        updatedMetadataList[rowIndex].key = value;
-        break;
-      case "type":
-        updatedMetadataList[rowIndex].type = value;
-        if (isObjectOrArray(value as any))
-          updatedMetadataList[rowIndex].value = "";
-        break;
-      case "value":
-        updatedMetadataList[rowIndex].value = value;
-        break;
-    }
+    updatedMetadataList[rowIndex][property] = value;
+    if (property === "type" && isObjectOrArray(value as any))
+      updatedMetadataList[rowIndex].value = "";
     setMetadataList(updatedMetadataList);
```

Comment:
I think this setMetadataList function should be outside of if condition so that state can update for all.

Reply:
It's already outside `if`